Created event check-in button and modal, added loading icon to modal

anniegw2 opened the pull request:
Refs #309

alan-fang commented on frontend/src/components/candidateCard.js:
```diff
@@ -56,30 +56,34 @@ class CandidateCardComponent extends Component {
-                    <Badge color="success">Accepted</Badge>
-                  </h5>
-                )}
+                {candidate.status &&
```
This seems overly boilerplate-d.. Is there any situation where `candidate.status` is not set?

alan-fang replied:
oh these are not even real changes lol, yeah please fix the version of `prettier` and re-request review!

tko22 commented:
> Generally, this looks good. However, you've passed the key into the request parameter; this isn't something I'll request changes to _now_, but I added a follow up issue (#313) for someone to resolve this issue in the entire `api.js` file.

key in the parameter is how we pass credentials to the backend. yes it’s not safe but it was a quick and easy way of providing “protected” to this application , which was necessary at that time. the new with stuff should fix all of that

alan-fang commented:
> > Generally, this looks good. However, you've passed the key into the request parameter; this isn't something I'll request changes to _now_, but I added a follow up issue (#313) for someone to resolve this issue in the entire `api.js` file.
> 
> key in the parameter is how we pass credentials to the backend. yes it’s not safe but it was a quick and easy way of providing “protected” info in this application, which was necessary at that time. the new with stuff should fix all of that

Did you mean new auth stuff? Yeah this sounds great, and when it's pushed I can close the ticket :)

alan-fang commented:
@anniegw2 is this a complete PR? Just saw that you pushed another commit here - generally if you're in the middle of doing work you can mark the PR as `WIP`

anniegw2 commented:
Refs #284 
Added check-in button on single event page: <img width="1041" alt="Screen Shot 2019-07-28 at 10 15 36 AM" src="https://user-images.githubusercontent.com/40253136/62009113-0f492600-b121-11e9-8bcc-7afdb60fa45f.png">
Added attendee check-in form: <img width="1034" alt="Screen Shot 2019-07-28 at 10 15 45 AM" src="https://user-images.githubusercontent.com/40253136/62009117-140dda00-b121-11e9-90d3-cd37eaec04e9.png">

alicesf2 commented on frontend/src/pages/event.js:
```diff
@@ -69,6 +95,19 @@ class Event extends Component {
             </h5>
```
can u get rid of this random </h5> lol

alicesf2 replied:
jk i'm dumb

alicesf2 commented on frontend/src/utils/api.js:
```diff
@@ -32,6 +32,21 @@ function createEvent(event) {
+    mode: 'cors'
+  }).then(res => res.json())
+}
+
```
oops sorry didn't catch this before but you have to export the api call at the bottom of the file, that should fix your error

alicesf2 commented:
![image](https://user-images.githubusercontent.com/32147571/62189229-50108d00-b323-11e9-98f4-cf501ef54198.png)

alan-fang commented:
not spinning 🤬

alicesf2 commented:
just imagine it pls

alicesf2 commented:
the id field in eventModal props is for use in the onReload function which is called after user checks in to event cuz it does all the api calls again to update the page and the eventId needs to be retained so it loads the event info correctly

arpanlaha commented on frontend/src/components/eventsModal.js:
```diff
@@ -45,39 +53,50 @@ class EventsModal extends Component {
 
   render() {
     const alert = <Alert color="danger">{this.state.alert}</Alert>
+    const loader = <ReactLoading className="loader" type="spinningBubbles" color="#000" />
```
Is there a reason we put loader and alert in variables rather than putting the literals where they appear? Looks like they're only used once.

alicesf2 replied:
good point lol

alicesf2 replied:
actually idk it looks cleaner to me but i can change it if u want

alicesf2 replied:
esp cuz they're used in the conditional rendering

alicesf2 replied:
ok nvm u win

arpanlaha commented on frontend/src/pages/event.js:
```diff
@@ -37,20 +41,44 @@ class Event extends Component {
+      year: this.state.year
+    }
+    const { success } = await eventCheckin(attendee, this.state.eventId)
+    return success
```
Why is success returned? Doesn't look like the return value of addAttendee is used anywhere.

alicesf2 replied:
it's used in the modal to display an error message if any fields are left blank
